protos/camera_server/camera_server.proto

```diff
@@ -143,3 +203,63 @@ message CameraServerResult {
     Result result = 1; // Result enum value
     string result_str = 2; // Human-readable English string describing the result
 }
+
+// Camera mode type.
+enum Mode {
+    MODE_UNKNOWN = 0; // Unknown mode
+    MODE_PHOTO = 1; // Photo mode
+    MODE_VIDEO = 2; // Video mode
+}
+
+// Information about the camera storage.
+message StorageInformation {
+    // Storage status type.
+    enum StorageStatus {
+        STORAGE_STATUS_NOT_AVAILABLE = 0; // Storage not available
+        STORAGE_STATUS_UNFORMATTED = 1; // Storage is not formatted (i.e. has no recognized file system)
+        STORAGE_STATUS_FORMATTED = 2; // Storage is formatted (i.e. has recognized a file system)
+        STORAGE_STATUS_NOT_SUPPORTED = 3; // Storage status is not supported
+    }
+
+    // Storage type.
+    enum StorageType {
+        STORAGE_TYPE_UNKNOWN = 0; // Storage type unknown
+        STORAGE_TYPE_USB_STICK = 1; // Storage type USB stick
+        STORAGE_TYPE_SD = 2; // Storage type SD card
+        STORAGE_TYPE_MICROSD = 3; // Storage type MicroSD card
+        STORAGE_TYPE_HD = 7; // Storage type HD mass storage
+        STORAGE_TYPE_OTHER = 254; // Storage type other, not listed
+    }
+
+    float used_storage_mib = 1; // Used storage (in MiB)
+    float available_storage_mib = 2; // Available storage (in MiB)
+    float total_storage_mib = 3; // Total storage (in MiB)
+
+    StorageStatus storage_status = 4; // Storage status
+
+    uint32 storage_id = 5; // Storage ID starting at 1
+
+    StorageType storage_type = 6; // Storage type
+
+    float read_speed = 7; // Read speed [MiB/s]
+    float write_speed = 8; // Write speed [MiB/s]
+}
+
+message CaptureStatus {
+    enum ImageStatus {
+        IMAGE_STATUS_IDLE = 0; // idle
+        IMAGE_STATUS_CAPTURE_IN_PROGRESS = 1; // capture in progress
+        IMAGE_STATUS_INTERVAL_IDLE = 2; // interval set but idle
+        IMAGE_STATUS_INTERVAL_IN_PROGRESS = 3; // interval set and capture in progress)
+    }
+    enum VideoStatus {
+        VIDEO_STATUS_IDLE = 0;  // idle
+        VIDEO_STATUS_CAPTURE_IN_PROGRESS = 1;   // capture in progress
+    }
+    float image_interval = 1; // Image capture interval (in s)
+    float recording_time_s = 2; // Elapsed time since recording started (in s)
+    float available_capacity = 3; // Available storage capacity. (in MiB)
```

Comment:
I suggest to add the units like the other fields:

```suggestion
    float available_capacity_mib = 3; // Available storage capacity. (in MiB)
```

And in other places too.